instrumentation/apache-pulsar/apache-pulsar-2.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/v28/telemetry/ConsumerAttributesExtractor.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pulsar.v28.telemetry;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import javax.annotation.Nullable;
+import org.apache.pulsar.client.api.Message;
+
+enum ConsumerAttributesExtractor implements AttributesExtractor<Message<?>, Attributes> {
+  INSTANCE;
+
+  private ConsumerAttributesExtractor() {}
+
+  @Override
+  public void onStart(AttributesBuilder attributes, Context parentContext, Message<?> message) {}
+
+  @Override
+  public void onEnd(
+      AttributesBuilder attributesBuilder,
+      Context context,
+      Message<?> message,
+      @Nullable Attributes attributes,
+      @Nullable Throwable error) {
+    if (null != attributes && !attributes.isEmpty()) {
```

Comment:
(just to conform with existing codebase)
```suggestion
    if (attributes != null && !attributes.isEmpty()) {
```